src/libman/Commands/BaseCommand.cs

```diff
@@ -216,5 +217,18 @@ protected void LogErrors(IEnumerable<IError> errors)
                 Logger.Log(string.Format("[{0}]: {1}", error.Code, error.Message), LogLevel.Error);
             }
         }
+
+        protected string GetUserInputWithDefault(string fieldName, string defaultFieldValue)
```

Comment:
This seems like it belongs on the `HostEnvironment` or the `IInputReader` implementation.  Whether the input is redirected is an environmental condition, right?  Feels a little odd to push that up to this level.

Why not have `ConsoleLogger` handle this within itself, since it is console specific?

Reply:
The only reason for doing the check here is to be able to show help message to the user before showing the error message. 

Other way is to throw an exception from the ConsoleLogger and catch that here and then show help.

Reply:
Are there other possible cases (aside from `CreateManifestAsync`) where we might hit the same issue with redirected input?  I.e. will we have to make this fix later in multiple places?

Reply:
Right now the only two possible scenarios are Init and Install command. 
Both are addressed.

The only other possibility is if in future libman allows multiple libraries with the same library name to be installed in the manifest (to resolve the library by asking the user to provide a choice).

Reply:
Along the lines of my comment below... what if I don't want a default provider?  Doesn't this just throw now?

Reply:
The current experience is designed to not allow manifests without a default provider. 
User can go ahead and delete it from the file, but the next install command will set whatever provider it uses to be the default one.

Reply:
Even if we implement #378, users will still face this issue if they don't pass in `-y` (or whatever we decide to call the option) on an input redirected console. 

I have moved the check to be handled by the `ConsoleLogger` which will throw if the input is redirected and the exception is being handled by the callers.